server/src/domain/metadata/metadata.service.ts

```diff
@@ -251,6 +252,39 @@ export class MetadataService {
     return true;
   }
 
+  async handleSidecarWrite(job: ISidecarWriteJob) {
+    const { id, description, dateTimeOriginal, latitude, longitude } = job;
+    const [asset] = await this.assetRepository.getByIds([id]);
+    if (!asset) {
+      return false;
+    }
+
+    const sidecarPath = asset.sidecarPath || `${asset.originalPath}.xmp`;
```

Comment:
Since we use this `'${asset.originalPath}.xmp'` in multiple places now, might be worth to have a function that either returns the sidecar path from the original path or always returns the sidecar path given an asset.

Reply:
I'm also fine tackling this after-the-fact.